DEVELOPERS.md

```diff
@@ -0,0 +1,420 @@
+
+If option 1 is preferred the source code for the component(s) to be modified has to be available on the filesystem, e.g. using `git clone`, before the next steps.
+
+### Creating an environment
+
+These instructions use [independent environments](https://spack-tutorial.readthedocs.io/en/latest/tutorial_environments.html#creating-an-independent-environment) rather than managed environments. The [main difference](https://spack-tutorial.readthedocs.io/en/latest/tutorial_environments.html#managed-versus-independent-environments) is that managed environments live within the spack directory structure, independent environments are outside it, and must be referenced directly by their path. 
+
+First step is choose a location for your development environment directory: it will only contain text files and possibly source code repositories. All compiled packages will be put in the directories defined in `spack-config` (defined with [`install_tree`](https://spack.readthedocs.io/en/latest/config_yaml.html#install-tree-root) in [`config.yaml`](https://github.com/ACCESS-NRI/spack-config/blob/main/common/config.yaml#L4)).
+
+Choose a name that suits the aim of the work. Make a directory with that name, `cd` into that directory then type the command `spack env create -d .`. This creates a `spack` environment with the same name as the directory. 
+
+### Modifying an environment
+
+Once the environment is created type `spack env activate .` from within the environment directory to activate the environment. 
+
+Add packages that are required, and use `spack develop` to indicate which packages will be modified and recompiled. 
+
+By default `spack` will clone the source code automatically, but it is possible to use an existing source code repository directory if that is preferred.
+
+It is necessary after `spack develop` to call `spack concretize -f` to force spack to update the concretization so that it picks up the changes to the packages that are being developed.
+
+## Compiling Modified Code
+
+Now the source code can be modified, and then compiled by invoking `spack install`
+
+## Examples
+
+The instructions above are best understood by example.
+
+### Developing a single model component
+
+In this example the `mom5` component will be modified but all other components and dependencies remain the same. 
+
+1. Create an environment with a name that reflects the purpose, activate, and set the `mom5` package to be a development environment: 
+```bash
+$ mkdir mom5-dev
+$ cd mom5-dev
+$ spack env create -d . ../../ACCESS-OM2/spack.yaml
+==> Created environment in /g/data/.../envs/mom5-dev
+==> You can activate this environment with:
+==>   spack env activate /g/data/.../envs/mom5-dev
+$ spack env activate .
+$ spack env status
+==> Using spack.yaml in current directory: /g/data/.../envs/mom5-dev
+$ spack develop mom5@git.master
+==> Configuring spec mom5@git.master=2023.11.09 for development at path mom5
+$ spack add mom5@git.2023.11.09
```

Comment:
Double check if you need to `add` a package. Since you have already added it via `develope`.